tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MatchTypeSuite.scala

```diff
@@ -0,0 +1,161 @@
+package scala.meta.tests.parsers.dotty
+
+import scala.meta.tests.parsers._
+import scala.meta._
+
+class MatchTypeSuite extends BaseDottySuite {
+
+  implicit val parseBlock: String => Stat = code => blockStat(code)(dialects.Dotty)
+
+  test("simple") {
+    val intput =
+      runTestAssert[Stat](
+        """|type Elem[X] = X match {
+           |  case String => Char
+           |  case Array[t] => t
+           |}
+           |""".stripMargin
+      )(
+        Defn.Type(
+          Nil,
+          Type.Name("Elem"),
+          List(Type.Param(Nil, Type.Name("X"), Nil, Type.Bounds(None, None), Nil, Nil)),
+          Type.Match(
+            Type.Name("X"),
+            List(
+              TypeCase(Type.Name("String"), Type.Name("Char")),
+              TypeCase(Type.Apply(Type.Name("Array"), List(Type.Name("t"))), Type.Name("t"))
+            )
+          )
+        )
+      )
+  }
+
+  test("tuple") {
+    runTestAssert[Stat](
+      """|type Head[X <: Tuple] = X match {
+         |  case (x1, ?) => x1
+         |}
+         |""".stripMargin
+    )(
+      Defn.Type(
+        Nil,
+        Type.Name("Head"),
+        List(
+          Type
+            .Param(Nil, Type.Name("X"), Nil, Type.Bounds(None, Some(Type.Name("Tuple"))), Nil, Nil)
+        ),
+        Type.Match(
+          Type.Name("X"),
+          List(
+            TypeCase(
+              Type.Tuple(
+                List(Type.Name("x1"), Type.Placeholder(Type.Bounds(None, None)))
+              ), // TODO should be Pat.TypeVar and Wildcard()
```

Comment:
.

Reply:
Forgot about that, I was thinking of adding additional nodes here, but might not actually make sense.

Reply:
Removed TODO, it doesn't make sens to add new trees here. Dotty syntax specifies it all as types not type pattern.